### DISCUSSION TITLE
Github hangs when I try to upload a profile picture

### DISCUSSION THREAD

**Discussion by brimy-pdx** [community/community#144144] [Other Feature Feedback, Questions, & Ideas] [answered]
Steps to reproduce:
* Go to https://github.com/settings/profile.
* Click Edit for Profile Picture.
* Click Upload a Photo.
* Browse to a photo file on my computer.
* Select one and click Open.
* A popup appears saying Crop your new profile picture. The body of the popup shows a spinning circle.

Expected: My picture should appear. (Or failing that, an error message.)
Actual: The circle keeps spinning and nothing further ever happens.

What I have tried:
- looking in the Console window for error messages
- uploading from Chrome, Edge, and the DuckDuckGo browser
- uploading into two different GitHub accounts
- turning VPN off and on
- uploading various images, all less than 1MB: JPG, GIF, PNG.
- uploading from an iOS device and a Windows device
- connecting to the internet through my ISP and through my phone

I feel entitled at this point to conclude the problem is not on my side, right? [](url)

**Comment by Cardosofiles** [comment, marked as answer]
Additional Troubleshooting Steps:
Clear Cache and Cookies: Sometimes, residual data can interfere with loading new content. Try clearing your cache and cookies for GitHub and then attempt the upload again.

Disable Browser Extensions: Some extensions can interfere with website functionality. Try uploading with all extensions disabled (or in an incognito/private browsing window with extensions off).

Resize the Image: GitHub may have limits on image file size or resolution for profile pictures. Try resizing the image to under 2MB or reducing the dimensions.

Check the File Format: GitHub might only accept specific formats like JPEG or PNG. Ensure your image is in one of these formats.

Attempt Upload from a Different Network: Even though you tried switching between your ISP and phone, testing from another network (like a public Wi-Fi) may provide additional insight.

Try a Different Image: If there’s a particular characteristic of the file (like high resolution), trying a simpler image file can help determine if it’s an issue with the specific image.

Reporting the Issue
If none of these steps resolve the problem, you’re well within reason to reach out to GitHub Support. Include details on all your troubleshooting steps, as they’ll help GitHub’s team understand the depth of the issue.

Here’s where to report it: [GitHub Support](https://support.github.com/).

This seems likely to be an issue on GitHub’s end, especially since you’ve tested across different browsers, devices, and accounts.

**Reply by brimy-pdx** [reply]
Thank you. I'll report it.

**Comment by redyf** [comment]
I can confirm this is also happening to me, even after clearing the cache and disabling all extensions. Trying other browsers (google-chrome, firefox, zen-browser) doesn't seem to do anything aswell.

**Comment by WillDaSilva** [comment]
I'll add that this is affecting me too, and nothing I've tried has fixed the issue.

**Comment by AardPlugs** [comment]
+1 Same Happens to me.

**Comment by l-fraga2811** [comment]
same here

**Comment by rthewitt** [comment]
Same issue.  I've tried all suggested solutions (deleted cache, cookies, local storage, without extensions, and incognito mode).  Firefox on Windows 11, uploading JPG.

**Comment by brimy-pdx** [comment]
Since I see others are also interested, you might like to know that GitHub responded to my report. They don't have a fix, but they are investigating:

-----
I have forwarded your report internally for further investigation. Although I currently don’t have an exact timeline for when this will be addressed, we'll update you when we have any news.

Note that while this support ticket may automatically close, the internal issue will remain open and under investigation.
 
We recommend that you keep an eye on our [Changelog](https://github.blog/changelog/) and our [GitHub Blog](http://github.blog/) where we publish updates on everything we ship.

**Comment by bee-san** [comment]
Thank you so much! Also have a similar problem.

**Comment by lebedevtech** [comment]
same to me

**Comment by prescott2m** [comment]
Same issue here. A workaround for now is to crop your image using a separate cropping tool, setting your GitHub profile picture to the cropped image, and just set it without cropping in GitHub. Works for me.

**Comment by brimy-pdx** [comment]
Not sure I understand. Are you saying that if you crop your image to some
set dimension in advance, then the upload skips the cropping stage and
simply works?

**Reply by prescott2m** [reply]
> Not sure I understand. Are you saying that if you crop your image to some set dimension in advance, then the upload skips the cropping stage and simply works?

Basically. If you crop your image to a square, the upload will still *try* to crop, but you can skip it by just pressing "Set new profile picture"

**Comment by 4wardx** [comment]
fixed

**Comment by brimy-pdx** [comment]
Actually now it's working for me. I didn't even get to try @winksplorer's workaround: the crop window works! I agree with @cnerdy: it appears to be fixed now.

**Comment by lebedevtech** [comment]
Yeah, for me it works too

**Comment by ghostinhershell** [comment]
Thanks for posting in the GitHub Community, @brimy-pdx!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the New to GitHub category is solely related to conversations around getting started on using GitHub and the community for beginners of GitHub, Community on Discussions, and those new to different areas of the product alike! This question should be in the Profile category. I've gone ahead and moved it for you. Good luck!

**Comment by Voxi0** [comment]
I'm still having this issue. But at least I can just click on set profile picture and skip cropping it and have it just as I want it.
